### ISSUE TITLE
`comma-spacing` doesn't handle sparse arrays

### ISSUE THREAD

**Issue by ljharb** [eslint/eslint#1587]
`[0, , 2]` triggers an error `There should be no space before ','` - in this case, the only non-whitespace character before the `,` in question is another `,`, so perhaps this should be an exception?

**Comment by jrvidal** [comment]
I'm looking into this.

**Comment by jrvidal** [comment]
Making progress.

* I'm confused by [this check](https://github.com/eslint/eslint/blob/master/lib/rules/comma-spacing.js#L77):
    ```javascript
    if (isSameLine(previousItemToken, commaToken) &&
                    isSameLine(commaToken, currentItemToken)
        ) {
            // Check spacing around a comma
        }
    ```

    We only check the spacing if the the comma, left and right tokens are in the same line. But shouldn't this be reported as invalid?

    ```javascript
    // comma-first mode, space after comma
    var x = 1
        ,y = 1;
    ```

* Shouldn't this rule include destructuring?
    ```javascript
    var [x ,y, ,z] = list;
    ```
    It should probably be a separate issue though.

**Comment by nzakas** [comment]
Yeah, seems like the first example should flag, and we should include destructuring. It might be easiest to rewrite the rule so it's more like `no-multi-spaces`, trying to find all commas through the AST is a bit precarious.

**Comment by jrvidal** [comment]
Ok, I'll take a look on `no-multi-spaces`.
